### PR TITLE
Board: Initial checklist support

### LINKED ISSUE #1594: Checklist support

### PR DESCRIPTION
**Pull request by annano**

### CODE REVIEW

File: packages/ui/src/components/EditWithIcon.svelte [hunk 1]
```diff
@@ -21,7 +21,7 @@
   import IconClose from './icons/Close.svelte'
   import plugin from '../plugin'
 
-  export let icon: Asset | AnySvelteComponent
```
Comment (haiodo) [inline packages/ui/src/components/EditWithIcon.svelte]: What the difference with EditBox if icon become optional?

File: plugins/board-resources/src/components/EditCard.svelte [hunk 2]
```diff
@@ -42,6 +43,19 @@
   let object: Card | undefined
   let state: State | undefined
   let handleMove: (e: Event) => void
+  let checklists: TodoItem[] = []
+
+  async function fetchChecklists () {
+    if (object) {
+      checklists = await client.findAll(task.class.TodoItem, { space: object.space, attachedTo: object._id })
```
Comment (haiodo) [inline plugins/board-resources/src/components/EditCard.svelte]: You should use live query

File: plugins/board-resources/src/components/editor/CardChecklist.svelte [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  const client = getClient()
+  let checklistItems: TodoItem[] = []
+  let done = 0
+  let editingName: string | undefined = undefined
+  let addingItemName: string | undefined = undefined
+  let hideDoneItems: boolean = false
+  let hovered: Ref<TodoItem> | undefined
+
+  async function fetch () {
+    checklistItems = await client.findAll(task.class.TodoItem, { space: value.space, attachedTo: value._id })
```
Comment (haiodo) [inline plugins/board-resources/src/components/editor/CardChecklist.svelte]: Please use live query

File: plugins/board-resources/src/components/editor/CardChecklist.svelte [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      assignee: null,
+      dueTo: null,
+      done: false
+    }
+    addingItemName = undefined
+    if (item.name.length <= 0) {
+      return
+    }
+    await client.addCollection(task.class.TodoItem, value.space, value._id, value._class, 'items', item)
+    fetch()
```
Comment (haiodo) [inline plugins/board-resources/src/components/editor/CardChecklist.svelte]: No need to fetch if live query is used

File: plugins/board-resources/src/components/editor/CardChecklist.svelte [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  async function setDoneToChecklistItem (item: TodoItem, event: CustomEvent<boolean>) {
+    const isDone = event.detail
+    if (!value) {
+      return
+    }
+    await client.update(item, { done: isDone })
+    fetch()
+  }
+
+  function showItemMenu (item: TodoItem, e?: Event) {
```
Comment (haiodo) [inline plugins/board-resources/src/components/editor/CardChecklist.svelte:94]: Please use Menu for view and regular actions contribution

File: plugins/board-resources/src/components/popups/AddChecklist.svelte [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+            assignee: item.assignee
+          })
+        )
+      )
+    }
+    dispatch('close')
+  }
+
+  client
+    .findAll(board.class.Card, { todoItems: { $gt: 0 } }, { lookup: { _id: { todoItems: task.class.TodoItem } } })
```
Comment (haiodo) [inline plugins/board-resources/src/components/popups/AddChecklist.svelte]: Please use asynchronous/await function

**Review by haiodo — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/EditWithIcon.svelte (hunk 1 above)
Comment (annano) [reply packages/ui/src/components/EditWithIcon.svelte]: Reverted

File: plugins/board-resources/src/components/EditCard.svelte (hunk 2 above)
Comment (annano) [reply plugins/board-resources/src/components/EditCard.svelte]: Done

File: plugins/board-resources/src/components/editor/CardChecklist.svelte (hunk 3 above)
Comment (annano) [reply plugins/board-resources/src/components/editor/CardChecklist.svelte]: Done

File: plugins/board-resources/src/components/editor/CardChecklist.svelte (hunk 4 above)
Comment (annano) [reply plugins/board-resources/src/components/editor/CardChecklist.svelte]: Done

File: plugins/board-resources/src/components/editor/CardChecklist.svelte (hunk 5 above)
Comment (annano) [reply plugins/board-resources/src/components/editor/CardChecklist.svelte:94]: Updated

File: plugins/board-resources/src/components/popups/AddChecklist.svelte (hunk 6 above)
Comment (annano) [reply plugins/board-resources/src/components/popups/AddChecklist.svelte]: Updated to `liveQuery`

**Review by haiodo — APPROVED** [review]

**Pull request merged**